### ISSUE TITLE
yaml.dump does not work with defaultdict

### ISSUE THREAD

**Issue by schrockn** [yaml/pyyaml#367]
Python 3.7.4
PyYAML 3.13

```
>>> yaml.dump(defaultdict(), default_flow_style=False)
'!!python/object/apply:collections.defaultdict []\n'
```

Nested ones can crash:

```
>>> dddict = defaultdict(lambda: defaultdict(dict))
>>> dddict['foo']['bar'] = 'baaz'
>>> yaml.dump(dddict, default_flow_style=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/__init__.py", line 200, in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/__init__.py", line 188, in dump_all
    dumper.represent(data)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/representer.py", line 27, in represent
    self.serialize(node)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/serializer.py", line 54, in serialize
    self.serialize_node(node, None, None)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/serializer.py", line 108, in serialize_node
    self.serialize_node(value, node, key)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/serializer.py", line 100, in serialize_node
    self.emit(SequenceEndEvent())
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/emitter.py", line 115, in emit
    self.state()
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/emitter.py", line 374, in expect_first_block_sequence_item
    return self.expect_block_sequence_item(first=True)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/emitter.py", line 384, in expect_block_sequence_item
    self.expect_node(sequence=True)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/emitter.py", line 242, in expect_node
    self.process_tag()
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/emitter.py", line 489, in process_tag
    self.prepared_tag = self.prepare_tag(tag)
  File "/Users/schrockn/venvs/prezi-dagster-dec-4/lib/python3.7/site-packages/yaml/emitter.py", line 607, in prepare_tag
    chunks.append('%%%02X' % ord(ch))
TypeError: ord() expected string of length 1, but int found
```

**Comment by perlpunk** [comment]
Can you try with the newest PyYAML? This has been fixed with version 5.1 and works for me.

**Comment by schrockn** [comment]
Confirmed it is fixed in 5.2. Thanks!

**Comment by udalrich** [comment]
This is failing for me with PyYaml 5.3.1 and Python 3.7.7.

```
$ python
Python 3.7.7 (default, Mar 10 2020, 15:43:03) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import yaml
>>> from collections import defaultdict
>>> yaml.dump(defaultdict(), default_flow_style=False)
'!!python/object/apply:collections.defaultdict []\n'
>>> dddict = defaultdict(lambda: defaultdict(dict))
>>> dddict['foo']['bar'] = 'baaz'
>>> yaml.dump(dddict, default_flow_style=False)
"!!python/object/apply:collections.defaultdict\nargs:\n- !!python/name:__main__.%3Clambda%3E ''\ndictitems:\n  foo: !!python/object/apply:collections.defaultdict\n    args:\n    - !!python/name:builtins.dict ''\n    dictitems:\n      bar: baaz\n"
>>> 
$ pip list
Package              Version Location                                            
-------------------- ------- ----------------------------------------------------
attrs                19.3.0  
coverage             5.1     
importlib-metadata   1.7.0   
more-itertools       8.4.0   
packaging            20.4    
pip                  20.0.2  
pluggy               0.13.1  
py                   1.9.0   
pyparsing            2.4.7   
pytest               5.4.3   
PyYAML               5.3.1   
setuptools           46.1.3  
six                  1.15.0  
wcwidth              0.2.5   
wheel                0.34.2  
zipp                 3.1.0   
```
